core/db/migrate/20220113052823_create_payment_sources.rb

```diff
@@ -0,0 +1,21 @@
+class CreatePaymentSources < ActiveRecord::Migration[5.2]
+  def change
+    create_table :spree_payment_sources do |t|
+      t.string :gateway_source_id
+      t.string :type
+
+      t.references :payment_method, index: true, foreign_key: { to_table: :spree_payment_methods }
+      t.references :user, index: true, foreign_key: { to_table: :spree_users }
+
+      if t.respond_to? :jsonb
+        t.jsonb :public_metadata
+        t.jsonb :private_metadata
+      else
+        t.json :public_metadata
+        t.json :private_metadata
+      end
+
+      t.timestamps
```

Comment:
@GrzegorzStochel let's add indexes for 

`type` &  `gateway_source_id` - shouldn't this be unique?

Reply:
we didn't have uniq on CC, but theoretically, it won't hurt to have that not to double the same records